src/cockpit/channels/stream.py

```diff
@@ -75,7 +76,11 @@ def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, A
             stderr = subprocess.PIPE
 
         env: Dict[str, str] = dict(os.environ)
-        env.update(options.get('env') or [])
+        if environ:
+            if not isinstance(environ, list):
```

Comment:
...but this makes it redundant.  This will fail if `environ` is `None`, so this check is enough on its own.